From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tony <tonymckahan@gmail.com>
Date: Sun, 15 Apr 2018 14:40:58 -0400
Subject: [ARCHEOLOGY] Rockchip next sync master and development

> X-Git-Archeology: > recovered message: > includes kernel panic fix on veth interfaces and audio device patch to enable Tinkerboard audio
> X-Git-Archeology: - Revision b1a21a44cde3d976b9b8ebce4f265e5aa79b2d8a: https://github.com/armbian/build/commit/b1a21a44cde3d976b9b8ebce4f265e5aa79b2d8a
> X-Git-Archeology:   Date: Sun, 15 Apr 2018 14:40:58 -0400
> X-Git-Archeology:   From: Tony <tonymckahan@gmail.com>
> X-Git-Archeology:   Subject: Rockchip next sync master and development
> X-Git-Archeology:
> X-Git-Archeology: - Revision b6f1503dfdb215429cc6dee6c2b94ca9ddc96fbc: https://github.com/armbian/build/commit/b6f1503dfdb215429cc6dee6c2b94ca9ddc96fbc
> X-Git-Archeology:   Date: Mon, 26 Nov 2018 01:38:25 -0500
> X-Git-Archeology:   From: Thomas McKahan <tonymckahan@gmail.com>
> X-Git-Archeology:   Subject: [ tinkerboard ] dev kernel add audio support
> X-Git-Archeology:
> X-Git-Archeology: - Revision 150ac0c2afa147d9e3b036c8ecd8238fe5648cf3: https://github.com/armbian/build/commit/150ac0c2afa147d9e3b036c8ecd8238fe5648cf3
> X-Git-Archeology:   Date: Tue, 19 Nov 2019 23:25:39 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Remove K<4, change branches, new features (#1586)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
> X-Git-Archeology: - Revision f08dcd48677d2a34f349bf571c979cd422bffcc3: https://github.com/armbian/build/commit/f08dcd48677d2a34f349bf571c979cd422bffcc3
> X-Git-Archeology:   Date: Tue, 31 Oct 2023 08:13:23 +0100
> X-Git-Archeology:   From: Paolo <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip,rk322x: bump edge kernel to 6.6 (#5875)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 562d96128ba6a511a8a06c0f4d29946ab80b8969: https://github.com/armbian/build/commit/562d96128ba6a511a8a06c0f4d29946ab80b8969
> X-Git-Archeology:   Date: Tue, 26 Dec 2023 16:45:30 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: consolidate rk322x and rockchip 32 bit families
> X-Git-Archeology:
> X-Git-Archeology: - Revision 54628d7d3e11824e560b77e905f69d52feb0fbd0: https://github.com/armbian/build/commit/54628d7d3e11824e560b77e905f69d52feb0fbd0
> X-Git-Archeology:   Date: Wed, 01 Jan 2025 19:38:55 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip: bump edge kernel to 6.13-rc5
> X-Git-Archeology:
> X-Git-Archeology: - Revision 7c55b4fce91f38383398a7498dde1c6d69a70495: https://github.com/armbian/build/commit/7c55b4fce91f38383398a7498dde1c6d69a70495
> X-Git-Archeology:   Date: Wed, 26 Mar 2025 22:23:29 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip 32bit edge kernel to v6.14
> X-Git-Archeology:
> X-Git-Archeology: - Revision cc4cb72d4069147ea1b5e6936de3b49aace21967: https://github.com/armbian/build/commit/cc4cb72d4069147ea1b5e6936de3b49aace21967
> X-Git-Archeology:   Date: Tue, 03 Jun 2025 09:53:37 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip 32 bit edge kernel to 6.15
> X-Git-Archeology:
> X-Git-Archeology: - Revision 96fe7dee19eaec6d9c5159a5cc50e33ca9c96096: https://github.com/armbian/build/commit/96fe7dee19eaec6d9c5159a5cc50e33ca9c96096
> X-Git-Archeology:   Date: Mon, 28 Jul 2025 20:45:52 +0800
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip edge to kernel 6.16
> X-Git-Archeology:
> X-Git-Archeology: - Revision 29317c6f7e33f2cc509acc0da23b615a7d9d8c31: https://github.com/armbian/build/commit/29317c6f7e33f2cc509acc0da23b615a7d9d8c31
> X-Git-Archeology:   Date: Thu, 18 Sep 2025 22:48:06 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: update rockchip 32 bit edge kernel to 6.17
> X-Git-Archeology:
---
 sound/usb/card.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/usb/card.c b/sound/usb/card.c
index 111111111111..222222222222 100644
--- a/sound/usb/card.c
+++ b/sound/usb/card.c
@@ -642,6 +642,14 @@ static void usb_audio_make_shortname(struct usb_device *dev,
 	}
 
 	strim(card->shortname);
+
+	/* Tinker Board ALC4040 CODEC	*/
+
+	if(USB_ID_VENDOR(chip->usb_id) == 0x0bda &&
+		USB_ID_PRODUCT(chip->usb_id) == 0x481a) {
+			strlcat(card->shortname, " OnBoard", sizeof(card->shortname));
+	}
+	
 }
 
 static void usb_audio_make_longname(struct usb_device *dev,
-- 
Armbian

